### PR TITLE
[System] Add journald support for auth and syslog data streams

### PR DESCRIPTION
**Pull request by belimawr**
## Note for reviewers

I did my best to get events that would test all the grok expressions, I mostly tried to get similar messages to the `syslog` and `auth` dataset. The `auth` dataset might be missing some from PAM because I wasn't sure how to generate them.

## Proposed commit message

This commit introduces support for Journald in the system integration,
adding a "journald" input for both auth and syslog data.

Conditions are used to select when to run the log input or the journal input, it
defaults to running the Journald input on Debian 12 and Amazon Linux 2023.

The minimum stack version is updated to `8.17.0` so the integration can use the
facilities filtering option from the journald input added in `8.16.0` and some extra fields
from the host provider added in `8.17.0`.

## Checklist

- [x] I have reviewed [tips for building integrations](https://github.com/elastic/integrations/blob/main/docs/tips_for_building_integrations.md) and this pull request is aligned with them.
- [x] I have verified that all data streams collect metrics or logs.
- [x] I have added an entry to my package's `changelog.yml` file.
- [x] I have verified that Kibana version constraints are current according to [guidelines](https://github.com/elastic/elastic-package/blob/master/docs/howto/stack_version_support.md#when-to-update-the-condition).
- [x] I have verified that any added dashboard complies with Kibana's [Dashboard good practices](https://docs.elastic.dev/ux-guidelines/data-viz/dashboard-good-practices) 

## Author's Checklist

- [x] Test on Windows
- [x] Ensure `data_stream.dataset` and `event.dataset` match their log input counterpart 
- [x] Wait for https://github.com/elastic/elastic-agent/pull/5941 before merging/releasing this integration version.

## How to test this PR locally
- Build the Elastic-Agent from `main`, you need the changes from (https://github.com/elastic/elastic-agent/pull/5941)
- Create an Elastic-Agent Policy
- Deploy the policy to your hosts.
- If the host is Debian 12 or Amazon Linux 2023, Journald inputs should be used for the auth and syslog data. Any other Linux host should use the log input.

## Related issues
- Closes https://github.com/elastic/integrations/issues/10797 

## Screenshots
<details><summary>Integration configuration</summary>
<p>

</p>
</details> 

<details><summary>Dashboards with data from Journald (Debain 12)</summary>
<p>

</p>
</details> 

<details><summary>Journald x Log input</summary>
<p>

I enabled both, the journald and the log input for the auth and syslog datasets on a fresh Debian 11 VM, both inputs collected the same amount of events. On the left is the journald input and on the right the log input.

</p>
</details>

### CODE REVIEW

File: packages/system/data_stream/auth_journald/agent/stream/journald.yml.hbs [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+# Workaround to have event.dataset and data_stream.dataset matching the
+# log input counterpart
+- add_fields:
+    target: data_stream
+    fields:
+      dataset: system.auth
+- add_fields:
+    target: event
+    fields:
+      dataset: system.auth
```
[the remark this answers is not in the document]
Comment (belimawr) [reply packages/system/data_stream/auth_journald/agent/stream/journald.yml.hbs:49]: > So the data stored in logs-system.auth_journald-default will have data_stream.dataset: system.auth?

[the remark this answers is not in the document]
Comment (belimawr) [reply packages/system/data_stream/auth_journald/agent/stream/journald.yml.hbs:49]: > So the data stored in logs-system.auth_journald-default will have data_stream.dataset: system.auth?

Yes, the goal here is to support Linux distros that moved away from traditional log files in favour of journald for system logs. This means keeping all the dashboards working (many, if not all of them filter on dataset, including user-built dashboards, saved searches, etc) while providing the correct monitoring for the Elastic-Agent.

If the `jouranld` input is used in the current datasets (`syslog` or `auth`), the Elastic-Agent will report the wrong input type on its overview/diagnostics/status, etc. It will also run the log input and journald input on the same Filebeat process. This happens because the Elastic-Agent reports and aggregates on the type declared in the data stream manifest, not the actual Beat input used.

Last, but not least, the data generated by the log and journald inputs are different, not completely, though. Thus this requires different ingest pipelines. One example is the the timestamp that is part of the journald entry, but has to be parsed from the message when ingesting a log file.

Setting the dataset as a Filebeat processor was the easiest way I found to meet all requirements we had.

Simply put: we need to support two different inputs, with different ingest pipelines in the same dataset.

File: packages/system/data_stream/auth_journald/agent/stream/journald.yml.hbs [hunk 2]
```diff
@@ -0,0 +1,49 @@
+type: journald
+
+facilities:
```
[the remark this answers is not in the document]
Comment (belimawr) [reply packages/system/data_stream/auth_journald/agent/stream/journald.yml.hbs:3]: > I would like to see a system test added that demonstrates the journald input config is working, and the filtering is selecting the proper logs.
> 
> In the [iptables](https://github.com/elastic/integrations/blob/0497b03a225be692198b75dc836f89a35f343713/packages/iptables/_dev/deploy/docker/docker-compose.yml#L14-L22) package we have a system test example that generates a journal file from logs stored in the [journald export](https://www.freedesktop.org/wiki/Software/systemd/export/) format (plain text).

I've tried it but it didn't work. I successfully managed to export the entries I needed and generated a small journal file, however the `elastic-agent` package would first start the Elastic-Agent container, then start the container with the journal file, because the input was already running and there was no journal file, it would fail to start. Currently the Journald input does not watch for files, it assumes they're there or it's reading from the system journal.

[the remark this answers is not in the document]
Comment (belimawr) [reply packages/system/data_stream/auth_journald/agent/stream/journald.yml.hbs:3]: I looked at the iptables test, it seems to be doing something similar to what I did, I'm a bit puzzled why mine didn't work... I'll try out the iptables one and see if I can get something working for the system integration.

File: packages/system/data_stream/auth_journald/agent/stream/journald.yml.hbs (hunk 1 above)
[the remark this answers is not in the document]
Comment (andrewkroh) [reply packages/system/data_stream/auth_journald/agent/stream/journald.yml.hbs:49]: > If the jouranld input is used in the current datasets (syslog or auth), the Elastic-Agent will report the wrong input type on its overview/diagnostics/status, etc. It will also run the log input and journald input on the same Filebeat process. This happens because the Elastic-Agent reports and aggregates on the type declared in the data stream manifest, not the actual Beat input used.

There are many integrations with multiple input types within one data stream (iptables being an example of one that uses both log file and journald). I have not observed the wrong input type being reported anywhere. And it runs each input in its own process. Is there something special about the system package that causes the problems you observed?

> Thus this requires different ingest pipelines.

[the remark this answers is not in the document]
Comment (belimawr) [reply packages/system/data_stream/auth_journald/agent/stream/journald.yml.hbs:49]: @andrewkroh thanks a lot for such a detailed explanation!!

> Is there something special about the system package that causes the problems you observed?
I don't believe so. I did not find anything in the documentation about it, so I implemented it the best way I could.


I'll dive deeper into the iptables and reproduce the setup in the system integration.

File: packages/system/data_stream/auth/_dev/test/pipeline/test-host-syslog-processor.json [hunk 3]
```diff
@@ -1,144 +1,156 @@
...
+    "events": [
+        {
+            "event": {
+                "original": "<165>1 2022-01-11T22:14:15.003Z mymachine.example.com eventslog 1024 ID47 [exampleSDID@32473 iut=\"3\" eventSource=\"Application\" eventID=\"1011\"][examplePriority@32473 class=\"high\"] this is the message"
             },
-            "severity": {
-              "code": 5,
-              "name": "Notice"
+            "@timestamp": "2022-01-11T22:14:15.003Z",
+            "input": {
```
Comment (belimawr) [inline packages/system/data_stream/auth/_dev/test/pipeline/test-host-syslog-processor.json:8]: Adding `input.type` is the only real change from this PR on this file, the rest is indentation from `elastic-package format`

**Comment by cmacknz** [comment]
We should:
1. Hide the default condition behind a "use journald" toggle, as it is likely to grow quite large over time. This also allows people to disable it and re-enable it without losing the condition.
2. We can have an advanced option for a custom condition that is empty by default.

**Comment by leehinman** [comment]
possible crazy thought.  What if we had an autodisover module that set a variable that for "system module input" to something like "log" or "journald".  Could the condition then use variable provided autodiscover?

**Comment by belimawr** [comment]
> possible crazy thought. What if we had an autodisover module that set a variable that for "system module input" to something like "log" or "journald". Could the condition then use variable provided autodiscover?

The biggest problem with that is that the Elastic-Agent needs to know the input type (that's defined in the integration manifest and does not need to map to a Beat input) before sending it to Filebeat. That's used to choose which OS process will run the input (e.g: one process for all journald inputs, another for all log inputs, etc), it is also used to aggregate and report status in multiple places, like the overview page for the Elastic-Agent. This also has implications on the state folder used.

Given all that, at the moment, we cannot postpone the decision about the input type, the input type needs to be known before the Elastic-Agent sends the configuration to Filebeat.

Another interesting challenge is that many things in Beats run "outside" of the input code, e.g: `input.type` is read/set based on the first configuration, a "proxy input" like the system-logs I wrote for supporting Debian 12 or even the container input will be handled as its own thing instead of the actual input that was instantiated and is running.

**Comment by belimawr** [comment]
> We should:
> 
> 1. Hide the default condition behind a "use journald" toggle, as it is likely to grow quite large over time. This also allows people to disable it and re-enable it without losing the condition.
> 2. We can have an advanced option for a custom condition that is empty by default.

I like this experience @cmacknz, however as far as I know it's not possible to do that with the current features we have on Fleet-UI. Maybe @kpollich can chime in and help understand what is supported by Fleet UI.

Clarifying more on this "use journald" toggle, does that mean having only one section for system logs, adding the "use journald" toggle that will show/hide a second level of "advanced options"
journald-integ


Instead of having the two configuration blocks (currently aggregated by input type) that's currently implemented on this PR:
two-blocks-foo

**Comment by leehinman** [comment]
> The biggest problem with that is that the Elastic-Agent needs to know the input type (that's defined in the integration manifest and does not need to map to a Beat input) before sending it to Filebeat.

Could that be addressed by having a "new" input type called "autodiscover"?

**Comment by belimawr** [comment]
> > The biggest problem with that is that the Elastic-Agent needs to know the input type (that's defined in the integration manifest and does not need to map to a Beat input) before sending it to Filebeat.
> 
> Could that be addressed by having a "new" input type called "autodiscover"?

Technically, yes. The question is more on a product side: How much we want to change the experience of running/managing Elastic-Agent?

Another more technical challenge is that the input type + output define which process will run that input, which also maps to the state folder, so changing input types (in the manifest) during an integration upgrade will make the state to be lost. If it's just on the Beats side, then the state is not an issue.

**Comment by cmacknz** [comment]
> We should:
> 
> - Hide the default condition behind a "use journald" toggle, as it is likely to grow quite large over time. This also allows people to disable it and re-enable it without losing the condition.
> - We can have an advanced option for a custom condition that is empty by default.

I managed to get the behavior I want with the patch below adding an `{{else}}` block, where there is a default condition that is hidden and you can add a custom condition that defaults to empty.

Screenshot 2024-11-13 at 4 18 06 PM

```diff
diff --git a/packages/system/data_stream/auth/agent/stream/journald.yml.hbs b/packages/system/data_stream/auth/agent/stream/journald.yml.hbs
index 48cfa38c82..e09d01a202 100644
--- a/packages/system/data_stream/auth/agent/stream/journald.yml.hbs
+++ b/packages/system/data_stream/auth/agent/stream/journald.yml.hbs
@@ -10,6 +10,8 @@ facilities:

 {{#if condition }}
 condition: {{condition}}
+{{else}}
+condition: ${host.os_version} == "12 (bookworm)" or (${host.os_platform} == "amzn" and ${host.os_version} == "2023")
 {{/if}}

 {{#if paths}}
diff --git a/packages/system/data_stream/auth/manifest.yml b/packages/system/data_stream/auth/manifest.yml
index 7315b0ef30..532739d81b 100644
--- a/packages/system/data_stream/auth/manifest.yml
+++ b/packages/system/data_stream/auth/manifest.yml
@@ -68,8 +68,7 @@ streams:
         type: text
         multi: false
         required: false
-        show_user: true
-        default: ${host.os_version} == "12 (bookworm)" or (${host.os_platform} == "amzn" and ${host.os_version} == "2023")
+        show_user: false

       - name: preserve_original_event
         required: true
diff --git a/packages/system/data_stream/syslog/agent/stream/journald.yml.hbs b/packages/system/data_stream/syslog/agent/stream/journald.yml.hbs
index a0d3a4cbd7..eb560cdf6c 100644
--- a/packages/system/data_stream/syslog/agent/stream/journald.yml.hbs
+++ b/packages/system/data_stream/syslog/agent/stream/journald.yml.hbs
@@ -18,6 +18,8 @@ facilities:

 {{#if condition }}
 condition: {{condition}}
+{{else}}
+condition: ${host.os_version} == "12 (bookworm)" or (${host.os_platform} == "amzn" and ${host.os_version} == "2023")
 {{/if}}

 {{#if paths}}
diff --git a/packages/system/data_stream/syslog/manifest.yml b/packages/system/data_stream/syslog/manifest.yml
index 03441b0a95..bfa5310799 100644
--- a/packages/system/data_stream/syslog/manifest.yml
+++ b/packages/system/data_stream/syslog/manifest.yml
@@ -74,8 +74,7 @@ streams:
         type: text
         multi: false
         required: false
-        show_user: true
-        default: ${host.os_version} == "12 (bookworm)" or (${host.os_platform} == "amzn" and ${host.os_version} == "2023")
+        show_user: false

       - name: preserve_original_event
         required: true
```

Also I noticed the condition is on each individual stream where it could be at the input level in the top level manifest instead of duplicating it everywhere. I think functionally it's the same it just clutters the agent policy more by repeating the condition unnecessarily. Inputs with no streams get removed.

**Comment by cmacknz** [comment]
The templated input I end up with looks exactly as I expect:

```yaml
inputs:
  - id: journald-system-e72572f5-bd7a-4a27-9f27-be40e81820f2
    name: system-1
    revision: 1
    type: journald
    use_output: default
    meta:
      package:
        name: system
        version: 1.63.0
    data_stream:
      namespace: default
    package_policy_id: e72572f5-bd7a-4a27-9f27-be40e81820f2
    streams:
      - id: journald-system.auth-e72572f5-bd7a-4a27-9f27-be40e81820f2
        type: journald
        data_stream:
          dataset: system.auth
          type: logs
        facilities:
          - 4
          - 10
        condition: >-
          ${host.os_version} == "12 (bookworm)" or (${host.os_platform} ==
          "amzn" and ${host.os_version} == "2023")
```

The condition comes and goes if I customize it as expected, but it does get surrounded with single quotes which seems like a bug to me.

Also nobody is going to have any idea what the possible values in the condition are unless we document them so writing a custom condition is a bit pointless, but at least we can hide it by default.

**Comment by belimawr** [comment]
Thanks Craig! I implemented it, added some extra tests and updated the screenshot in the PR description.

**Comment by belimawr** [comment]
@andrewkroh, the PR is ready for review again.

**Comment by jlind23** [comment]
@lalit-satapathy can we get someone from your team review it too please?

**Comment by jlind23** [comment]
@belimawr shouldn't this version only be available from 8.17.0 as only this one will include the latest changes on Beats for the journald input?

**Comment by belimawr** [comment]
> @belimawr shouldn't this version only be available from 8.17.0 as only this one will include the latest changes on Beats for the journald input?

Good catch @jlind23! We're using a filtering option that was added in `8.16.0`. I'll update it.

**Review by andrewkroh — COMMENTED** [review]
Looks good, just a few minor comments. I did not review in detail the contents of the new pipelines under the assumption that they are close derivatives of the existing ones.

I do think some documentation should be added to the README to explain a bit of how it works (or how it is supposed to work) w.r.t. using journald on newer OSes and that this is based on the condition.

File: packages/system/manifest.yml [hunk 4]
```diff
@@ -30,7 +30,26 @@ policy_templates:
...
+        title: Collect logs from System instances using Journald
+        description: Collecting System auth and syslog logs using Journald
+        vars:
+          - name: condition
+            title: Condition
+            description: Condition to filter when to apply this datastream. Refer to [Host provider](https://www.elastic.co/guide/en/fleet/current/host-provider.html) to find the available keys and to [Conditions](https://www.elastic.co/guide/en/fleet/current/dynamic-input-configuration.html#conditions) on how to use the available keys in conditions.
+            type: text
+            multi: false
+            required: false
+            show_user: false
```
[the remark this answers is not in the document]
Comment (cmacknz) [reply packages/system/manifest.yml:52]: No reason we can't pre-populate the custom condition with the default

File: packages/system/data_stream/auth/elasticsearch/ingest_pipeline/journald.yml [hunk 5]
```diff
@@ -0,0 +1,353 @@
...
+        - syslog
+        - systemd
+        - message_id
+on_failure:
+  - set:
+      field: event.kind
+      value: pipeline_error
+  - set:
+      field: error.message
+      value: "Processor {{ _ingest.on_failure_processor_type }} with tag {{ _ingest.on_failure_processor_tag }} in pipeline {{ _ingest.on_failure_pipeline }} failed with message: {{ _ingest.on_failure_message }}"
```
[the remark this answers is not in the document]
Comment (belimawr) [reply packages/system/data_stream/auth/elasticsearch/ingest_pipeline/journald.yml]: Done on 9f2a41f86e

File: packages/system/manifest.yml (hunk 4 above)
[the remark this answers is not in the document]
Comment (belimawr) [reply packages/system/manifest.yml:52]: I've realised something, while the current approach works very well for the use case we designed it (Debian 12/Amazon Linux 2023, etc), we also made it impossible for a user to set no conditions on the journald and log inputs.

I believe the ideal solution would be to add some UI feature to allow fields to be shown/hidden based on the state of the others. I can see other integrations benefiting from this feature as well.

[the remark this answers is not in the document]
Comment (belimawr) [reply packages/system/manifest.yml:52]: A quick win option would be to have the default value being our , but I agree the user experience will be far from ideal and it can be very confusing.

**Comment by belimawr** [comment]
> I do think some documentation should be added to the README to explain a bit of how it works (or how it is supposed to work) w.r.t. using journald on newer OSes and that this is based on the condition.

I added something on a9731b80c2, I tried to keep it generic as the current implementation will likely change a bit.

File: packages/system/manifest.yml (hunk 4 above)
[the remark this answers is not in the document]
Comment (belimawr) [reply packages/system/manifest.yml:52]: We could also get close with the current features by having a toggle that means "use default conditions" and if it is selected, then the conditions field has no effect, if it's not selected, then we render the condition field, allowing the user to define their own conditions.

@andrewkroh do you know if it is possible to do validation on the fields based on other field's value? Having zero UI feedback between the interaction of the toggle and the condition field is far from ideal, an error/validation message would at least prevent configurations that have no effect.

**Review by ishleenk17 — APPROVED** [review]
Code owner's approval!

File: packages/system/manifest.yml (hunk 4 above)
[the remark this answers is not in the document]
Comment (andrewkroh) [reply packages/system/manifest.yml:52]: I'm not aware of any Fleet feature that would allow this input validation. Would be nice to have.

File: .github/CODEOWNERS [hunk 6]
```diff
@@ -374,6 +375,7 @@
 /packages/system/data_stream/process_summary @elastic/obs-infraobs-integrations
 /packages/system/data_stream/socket_summary @elastic/obs-infraobs-integrations
 /packages/system/data_stream/syslog @elastic/obs-infraobs-integrations
+/packages/system/data_stream/syslog_journald @elastic/obs-infraobs-integrations
```
[the remark this answers is not in the document]
Comment (belimawr) [reply .github/CODEOWNERS]: Yes, that's correct.

A previous version of this PR had them as new datastreams and I forgot to remove the entries in the CODEOWNERS. Thanks for spotting that! It's fixed by 91d4fd08b0

File: packages/system/manifest.yml (hunk 4 above)
[the remark this answers is not in the document]
Comment (belimawr) [reply packages/system/manifest.yml:52]: I believe many integrations would benefit from that. I recall seeing very un-intuitive logic in the hbs files that, unfortunately, are not reflected in the UI and have lead to SDHs. Adding a feature like that would really improve user experience.

[the remark this answers is not in the document]
Comment (cmacknz) [reply packages/system/manifest.yml:52]: > we also made it impossible for a user to set no conditions on the journald and log inputs.

Could they not just set a custom condition that is always true, like host.platform == linux?

File: packages/system/manifest.yml [hunk 7]
```diff
@@ -1,14 +1,14 @@
...
-version: "1.62.1"
+version: "1.63.0"
 description: Collect system logs and metrics from your servers with Elastic Agent.
 type: integration
 categories:
   - os_system
 conditions:
   kibana:
-    version: "^8.13.0"
+    version: "^8.16.0"
```
Comment (cmacknz) [inline packages/system/manifest.yml:11]: What happens if you remove this? What do older agent versions do if they don't support the new host.* parameters for the journald condition?

Doing this means users won't get bug fixes for the integration unless they upgrade to 8.16.0. I want to double check that this is actually solving a problem that can't instead be solved by documentation (e.g. that journald conditions don't work until this version).

Comment (cmacknz) [reply packages/system/manifest.yml:11]: Any chance we can put in a condition on the agent version to get around this?

Comment (belimawr) [reply packages/system/manifest.yml:11]: So, I did some testing and the results are interesting:
- The conditions silently break, so the log input runs on Debian 12 and the journald does not.
- Because the filters do not work, both datasets (auth and syslog) ingest all the data
- The dashboards seem to work well, because they use specific fields set by the ingest pipelines, I believe their data is accurate.

So, overall, it does not seem to bad, we duplicate data, but that's about it.



> Any chance we can put in a condition on the agent version to get around this?

That's a good idea, I'll check. However it's gonna be a bit unclear for the user why the journald input shows as not available.

Comment (belimawr) [reply packages/system/manifest.yml]: I've updated the PR with the latest changes and the description with the new screenshot

<details><summary>new screenshot</summary>
<p>




</p>
</details>

**Comment by belimawr** [comment]
Folks, that's the final version. We've reverted to using the conditions default value and documenting it.

**Review by andrewkroh — APPROVED** [review]

File: packages/system/manifest.yml (hunk 7 above)
Comment (cmacknz) [reply packages/system/manifest.yml]: > So, overall, it does not seem to bad, we duplicate data, but that's about it.

We duplicate data on all non-8.16.0 agents? This is going to be a very large number of agents, and this integration is installed by default, so if there's any consequence or effect to removing the version constraint like this let's put the constraint back. I'm confident we did our diligence that it has to be there.

The system integration is the first integration most users interact with, it can't have any quirks or weird behavior or duplicate anything for currently supported 8.x versions (which is more than 8.16.x).

File: packages/system/manifest.yml [hunk 8]
```diff
@@ -30,7 +30,40 @@ policy_templates:
...
+              Condition to filter when to apply this input. Refer to
+              [Host provider](https://www.elastic.co/guide/en/fleet/current/host-provider.html)
+              to find the available keys and to
+              [Conditions](https://www.elastic.co/guide/en/fleet/current/dynamic-input-configuration.html#conditions)
+              on how to use the available keys in conditions. It defaults to
+              '${host.os_version} != "12 (bookworm)" and (${host.os_platform} != "amzn" or ${host.os_version} != "2023")'
+            type: text
+            multi: false
+            required: false
+            show_user: false
```
Comment (cmacknz) [inline packages/system/manifest.yml:47]:
```suggestion
            show_user: true
```

File: packages/system/manifest.yml [hunk 9]
```diff
@@ -30,7 +30,40 @@ policy_templates:
...
+              Condition to filter when to apply this input. Refer to
+              [Host provider](https://www.elastic.co/guide/en/fleet/current/host-provider.html)
+              to find the available keys and to
+              [Conditions](https://www.elastic.co/guide/en/fleet/current/dynamic-input-configuration.html#conditions)
+              on how to use the available keys in conditions. It defaults to
+              '${host.os_version} == "12 (bookworm)" or (${host.os_platform} == "amzn" and ${host.os_version} == "2023")'
+            type: text
+            multi: false
+            required: false
+            show_user: false
```
Comment (cmacknz) [inline packages/system/manifest.yml:65]:
```suggestion
            show_user: true
```

**Review by cmacknz — COMMENTED** [review]
The only two changes I want are to put the stack version constraint back, and to show the user that the condition exists by default. Let's not hide the fact that the inputs are conditional.

I think the documentation for the condition is great, and the simplification to have the condition apply to the whole input instead of every stream makes things easier to see.

File: packages/system/manifest.yml (hunk 7 above)
Comment (belimawr) [reply packages/system/manifest.yml]: > We duplicate data on all non-8.16.0 agents?

No, that's only for Elastic-Agents where the Jouranld input ends up running, the user would have to change the conditions to get the Journald input running on agents < 8.16.0.

So, by default, there will be no issue, however this does introduce a weird behaviour, which is undocumented and un-intuitive.

I'll revert to require at least 8.16.0

**Review by cmacknz — APPROVED** [review]
Thanks!

Remember that once you merge this it will be available to all 8.16+ stacks rapidly afterwards
